crates/bevy_reflect/src/serde/de.rs

```diff
@@ -380,130 +397,194 @@ impl<'a, 'de> Visitor<'de> for MapVisitor<'a> {
     }
 }
 
-struct StructDeserializer<'a> {
-    registry: &'a TypeRegistry,
-}
-
-impl<'a, 'de> DeserializeSeed<'de> for StructDeserializer<'a> {
-    type Value = DynamicStruct;
-
-    fn deserialize<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        deserializer.deserialize_map(StructVisitor {
-            registry: self.registry,
-        })
-    }
-}
-
-struct StructVisitor<'a> {
-    registry: &'a TypeRegistry,
-}
-
-impl<'a, 'de> Visitor<'de> for StructVisitor<'a> {
-    type Value = DynamicStruct;
-
-    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
-        formatter.write_str("struct value")
-    }
-
-    fn visit_map<V>(self, mut map: V) -> Result<Self::Value, V::Error>
-    where
-        V: MapAccess<'de>,
-    {
-        let mut dynamic_struct = DynamicStruct::default();
-        while let Some(key) = map.next_key::<String>()? {
-            let value = map.next_value_seed(ReflectDeserializer {
-                registry: self.registry,
-            })?;
-            dynamic_struct.insert_boxed(&key, value);
-        }
-
-        Ok(dynamic_struct)
-    }
-}
-
-struct TupleStructDeserializer<'a> {
-    registry: &'a TypeRegistry,
-}
-
-impl<'a, 'de> DeserializeSeed<'de> for TupleStructDeserializer<'a> {
-    type Value = DynamicTupleStruct;
-
-    fn deserialize<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        deserializer.deserialize_seq(TupleStructVisitor {
-            registry: self.registry,
-        })
-    }
-}
-
-struct TupleStructVisitor<'a> {
-    registry: &'a TypeRegistry,
-}
-
-impl<'a, 'de> Visitor<'de> for TupleStructVisitor<'a> {
-    type Value = DynamicTupleStruct;
-
-    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
-        formatter.write_str("tuple struct value")
-    }
-
-    fn visit_seq<V>(self, mut seq: V) -> Result<Self::Value, V::Error>
-    where
-        V: SeqAccess<'de>,
-    {
-        let mut tuple_struct = DynamicTupleStruct::default();
-        while let Some(value) = seq.next_element_seed(ReflectDeserializer {
-            registry: self.registry,
-        })? {
-            tuple_struct.insert_boxed(value);
-        }
-        Ok(tuple_struct)
-    }
-}
-
-struct TupleDeserializer<'a> {
-    registry: &'a TypeRegistry,
-}
-
-impl<'a, 'de> DeserializeSeed<'de> for TupleDeserializer<'a> {
-    type Value = DynamicTuple;
-
-    fn deserialize<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        deserializer.deserialize_seq(TupleVisitor {
-            registry: self.registry,
-        })
-    }
-}
-
 struct TupleVisitor<'a> {
```

Comment:
Extreme tinny nit: I would move `TupleVisitor` impl just after the `TupleStructVisitor` impl, so that it's obvious the code for the two is very similar, and they are easier to identify.